### PR TITLE
Major update to terraform 0.13

### PR DESCRIPTION
**Pull request by jecnua**
- Tested with 1.19.x
- Tested with terraform 0.13.4
- Migrated cgroup driver to systemd
- Using Calico as default CNI (but made the choice parametric).
- Enabling access to the cluster nodes via SSM (now the ec2 key is optional)
- Adding fixes for CIS benchmark
- More flexibility while setting up the control plane.
- Both workers and control plane are not using kubeadm config to init

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 0.13.4
+    - run: |
+        cd examples/tf-0-13
+        terraform init
+        terraform fmt -check
```
Comment (adamwshero) [inline .github/workflows/tests.yml:72]: I see the new line warning here. (shrug)

File: README.md [hunk 2]
```diff
@@ -13,11 +13,12 @@ environment. More informations can be found at its own repo:
 - [Kubernetes module](modules/kubernetes/)
 - - [Module maintainers](MAINTAINERS.md)
 
-*NOTE*: It only suppport terraform 0.12.x onward. 
+*NOTE*: It only suppport terraform 0.13.x onward.
```
Comment (adamwshero) [inline README.md:17]: Just some spelling and things... *NOTE*: It only supports Terraform 0.13.x onward.

File: modules/kubernetes/00-data_sources.tf [hunk 3]
```diff
@@ -0,0 +1 @@
+data "aws_partition" "current" {}
```
Comment (adamwshero) [inline modules/kubernetes/00-data_sources.tf:1]: another annoying newline warning. :-/

File: modules/kubernetes/00-variables_defaults.tf [hunk 4]
```diff
@@ -79,11 +79,50 @@ variable "sns_topic_notifications" {
...
+  type        = string
+  description = "The key name to associate to the new ec2 servers. Not needed if you use SSM or want no access"
+  default     = ""
+}
+
+variable "enable_admission_plugins" {
+  type        = string
+  description = "The comma separated list of admission plugin to enable"
+  default     = "NodeRestriction" # Def for 1.19
+}
```
Comment (adamwshero) [inline modules/kubernetes/00-variables_defaults.tf:128]: :-/

File: modules/kubernetes/05-iam.tf [hunk 5]
```diff
@@ -43,13 +44,19 @@ resource "aws_iam_role" "k8s_assume_role" {
...
 EOF
-
 }
+
+resource "aws_iam_role_policy_attachment" "ssm_policy_att" {
+  count      = var.enable_ssm_access_to_nodes ? 1 : 0
+  depends_on = [aws_iam_role.k8s_assume_role]
+  role       = aws_iam_role.k8s_assume_role.name
+  policy_arn = "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AmazonEC2RoleforSSM"
+}
```
Comment (adamwshero) [inline modules/kubernetes/05-iam.tf:62]: :-D

File: modules/kubernetes/scripts/kubeadm_config.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    audit-log-maxage: "30"
+    audit-log-maxbackup: "10"
+    audit-log-maxsize: "100"
+    kubelet-certificate-authority: "/etc/kubernetes/pki/ca.crt"
+controllerManager:
+  extraArgs:
+    profiling: "false"
+scheduler:
+  extraArgs:
+    profiling: "false"
```
Comment (adamwshero) [inline modules/kubernetes/scripts/kubeadm_config.yaml:23]: :-o

**Review by adamwshero — APPROVED** [review]
Looks good and thorough Fab. ..just a few nit pick things.

File: README.md (hunk 2 above)
Comment (jecnua) [reply README.md:17]: Thanks, fixed.

**Pull request merged**